server.go

```diff
@@ -1487,21 +1492,29 @@ func (s *Server) ReadSeries(database, retentionPolicy, name string, tags map[str
 	return values, nil
 }
 
-func (s *Server) ExecuteQuery(q *influxql.Query, database string, user *User) (interface{}, error) {
-	for _, stmt := range q.Statements {
-		switch c := stmt.(type) {
+// ExecuteQuery executes an InfluxQL query against the server.
+// Returns a resultset for each statement in the query.
+// Stops on first execution error that occurs.
+func (s *Server) ExecuteQuery(q *influxql.Query, database string, user *User) Results {
+	// Build empty resultsets.
+	results := make(Results, len(q.Statements))
+
+	// Execute each statement.
+	for i, stmt := range q.Statements {
+		var res *Result
+		switch stmt := stmt.(type) {
```

Comment:
This is hilarious. When I orignally coded this I also used "stmt"' but had assumed it would confuse the compiler. But it seems liked it's ok to use "stmt" at these scopes?
.